### PR TITLE
Scan: Implemented the sort function for validator homepage

### LINKED ISSUE #2321: Sorting selection on Validator Home Page (mobile)

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #2321

## Implementation details

 - Implemented the sorting dropdown on the validator homepage's mobile layout.
 - Fixed scroll issue when page redirection.
 - CSS refactoring a bit.
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/pages/ValidatorHomePage.re [hunk 1]
```diff
@@ -91,6 +161,61 @@ let getCurrentDay = _ => {
...
+      <div className={Styles.sortDrowdownPanel(show)}>
+        {sortList
+         ->Belt.List.map(value => {
+             let isActive = sortedBy == value;
+             <div
+               className={Styles.sortDropdownItem(isActive)}
+               onClick={_ => {
+                 setSortedBy(_ => value);
+                 setShow(_ => false);
+               }}>
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: Please add key here.

File: scan/src/components/ValidatorsTable.re [hunk 2]
```diff
@@ -253,6 +253,28 @@ type sort_by_t =
...
+let getDirection =
+  fun
+  | NameAsc
+  | VotingPowerAsc
+  | CommissionAsc
+  | UptimeAsc => "ASC"
+  | NameDesc
+  | VotingPowerDesc
+  | CommissionDesc
+  | UptimeDesc => "DESC";
```
Comment (evilpeach) [inline scan/src/components/ValidatorsTable.re]: Let's declare it to type.

```
type sort_direction_t = ASC | DESC;
```

File: scan/src/images/Images.re [hunk 3]
```diff
@@ -66,6 +66,8 @@
 [@bs.module] external ledgerStep2Cosmos: string = "./ledger-step2-cosmos.svg";
 [@bs.module] external ledgerStep2BandChain: string = "./ledger-step2-bandchain.svg";
 [@bs.module] external sort: string = "./sort.svg";
+[@bs.module] external mobileSort: string = "./mobile-sort-icon.svg";
+[@bs.module] external mobileSortActive: string = "./mobile-sort-icon-active.svg";
```
Comment (evilpeach) [inline scan/src/images/Images.re]: Do they mean Asc and Desc? If they do, let's use `mobileSortAsc` and `mobileSortDesc` name.

File: scan/src/pages/ValidatorHomePage.re [hunk 4]
```diff
@@ -69,15 +78,76 @@ module Styles = {
...
+  let sortImage = direction => {
+    style([
+      transform(
+        scaleY(
+          {
+            direction === "ASC" ? 1 |> float_of_int : (-1) |> float_of_int;
+          },
+        ),
+      ),
+    ]);
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: Switch on `sort_direction_t`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/images/Images.re (hunk 3 above)
Comment (pzshine) [reply scan/src/images/Images.re]: Nope, mobileSort for normal state (grey color) and mobileSortActive for active state(blue color).

File: scan/src/pages/ValidatorHomePage.re (hunk 1 above)
Comment (pzshine) [reply scan/src/pages/ValidatorHomePage.re]: I'm not sure I can whether use the` index` of mapWithIndex or not?

File: scan/src/components/ValidatorsTable.re [hunk 5]
```diff
@@ -253,6 +256,28 @@ type sort_by_t =
...
+let getName =
+  fun
+  | NameAsc => "Validator name"
+  | VotingPowerAsc => "Voting Power"
+  | CommissionAsc => "Commission"
+  | UptimeAsc => "Uptime"
+  | NameDesc => "Validator name"
+  | VotingPowerDesc => "Voting Power"
+  | CommissionDesc => "Commission"
+  | UptimeDesc => "Uptime";
```
Comment (evilpeach) [inline scan/src/components/ValidatorsTable.re]: Let's change to 
```
let getName =
    fun
    | NameAsc
    | NameDesc => "Validator Name"
    | VotingPowerAsc
    | VotingPowerDesc => "Voting Power"
...
```

File: scan/src/pages/ValidatorHomePage.re [hunk 6]
```diff
@@ -91,6 +154,75 @@ let getCurrentDay = _ => {
...
+        {sortList
+         ->Belt.List.mapWithIndex((i, value) => {
+             let isActive = sortedBy == value;
+             <div
+               key={i |> string_of_int}
+               className={Styles.sortDropdownItem(isActive)}
+               onClick={_ => {
+                 setSortedBy(_ => value);
+                 setShow(_ => false);
+                 Js.Console.log(value);
```
Comment (evilpeach) [inline scan/src/pages/ValidatorHomePage.re]: remove

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**